### PR TITLE
[Middleware_02] Define MiddlewareCommDatabaseClassTests

### LINKED ISSUE #70: [Middleware_02] Build and Test Middleware_02 Production Classes
**What is the use case? Please describe.**
For this sprint task, you'll be building and testing all of the classes in the `edu.oakland.production.middleware02` package. Note: This issue is similar to #64 , but is tracked differently as helper and production classes have different deliverable dates.

**Sprint Task Setup**

To be done by @mckenzieking , @kwillard34 , or @rcapollari before the rest of the team starts working on it:

1. Edit this issue message to add a checkbox for each class, and assign it to the correct team member from #59 .

**Prerequisites**

Please makes sure that your test plan from #59 for the class you are writing is closed and merged to main before starting this, to ensure that all of the documentation you need for this is in the correct spot.

**Steps**

To be done by each team member for each class they are developing a test plan for:

1. Each team member should watch the last ~18 minutes of @andrewdimmer 's [Test Driven Development Crash Course](https://youtu.be/eSc0aY4GhPs)
2. One person should create a branch off of main called `define-Name_Of_Class` (ex. the `Demo` classes and tests get build on a branch called `build-Demo`). Everyone else in the programming pair or group should then checkout that branch.
3. Using the information in the video, use test driven development to build all the required classes, stubs, and tests. Remember: you've already defined what you need to build at https://github.com/andrewdimmer/csi-3370-software-project/tree/main/docs/uml-class-diagrams/middleware02 !
4. Commit your changes with the message `[Middleware_02] Built Name_Of_Class`
5. Push your changes
6. Create a pull request with the name `[Middleware_02] Built Name_Of_Class` and the rest of the information like description, tags, etc. For reference, the settings on the right hand side should match this issue, with the exception that the assignees on the PR are just those who are pushing code.
7. Once your code is complete, add @andrewdimmer as a reviewer.

**Tasks to Complete**

- [x] Allocate Team Members to Classes - @mckenzieking , @kwillard34 , or @rcapollari
- [x] Build MiddlewareCommInterfaceImplementation - @trode21 and @NMoua-Edu
- [ ] Build MiddlewareCommLinkManagerImplementation - @tomtomtomcat and @nbergamin123
- [x] Build MiddlewareCommDatabaseInterfaceImplementation - @rcapollari and @GitLee32

**Additional context**
Note: if for some reason you need to change a public interface or shared data type, reach out to @andrewdimmer who can help you manage making that BREAKING CHANGE.

Note: if you run into merge conflicts, let @andrewdimmer know and he can help you resolve them.

### PR DESCRIPTION
**Pull request by kwillard34**
working on #70

### CODE REVIEW

File: docs/uml-class-diagrams/middleware02/test/MiddlewareCommDatabaseInterfaceImplementationTests/MiddlewareCommDatabaseInterfaceImplementationTests.puml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 package edu.oakland.test.middleware02 {
     class MiddlewareCommDatabaseInterfaceImplementationTests {
-
+      + void RfidInIsRfidOutTest()
```
Comment (andrewdimmer) [inline docs/uml-class-diagrams/middleware02/test/MiddlewareCommDatabaseInterfaceImplementationTests/MiddlewareCommDatabaseInterfaceImplementationTests.puml]: Fix the casing --> Methods in Java are always in camelCase.
Also, make sure to have a test case to verify that the `DatabaseCommInterface` is not null!
```suggestion
        + void rfidInIsRfidOut()
        + void databaseCommInterfaceIsNotNull()
```

**Review by andrewdimmer — CHANGES_REQUESTED** [review]
One minor change just to make sure you test the constructor, then you're good to go!

**Review by kwillard34 — COMMENTED** [review]
Needs Review

**Review by kwillard34 — COMMENTED** [review]
needs review

File: docs/uml-class-diagrams/middleware02/test/MiddlewareCommLinkManagerImplementationTests/MiddlewareCommLinkManagerImplementationTests.puml [hunk 2]
```diff
@@ -2,7 +2,10 @@
```
Comment (andrewdimmer) [inline docs/uml-class-diagrams/middleware02/test/MiddlewareCommLinkManagerImplementationTests/MiddlewareCommLinkManagerImplementationTests.puml:2]: Can we remove this file from this PR? I think that this is the one from #96 right?

**Review by andrewdimmer — CHANGES_REQUESTED** [review]

**Review by kwillard34 — COMMENTED** [review]
Needs review for deletion

File: docs/uml-class-diagrams/middleware02/test/MiddlewareCommLinkManagerImplementationTests/MiddlewareCommLinkManagerImplementationTests.puml [hunk 3]
```diff
@@ -2,7 +2,10 @@
 
 package edu.oakland.test.middleware02 {
     class MiddlewareCommLinkManagerImplementationTests {
-
+      + void verifyTrackLengthTest()
+      + void rfidInIsRfidOut()
+      + void gpsDataInIsGpsDataOut()
+      + void trackDataInIsTrackDataOut()
```
Comment (andrewdimmer) [inline docs/uml-class-diagrams/middleware02/test/MiddlewareCommLinkManagerImplementationTests/MiddlewareCommLinkManagerImplementationTests.puml]:
```suggestion

```

**Review by andrewdimmer — APPROVED** [review]
Should be good to merge!

**Review by mckenzieking — APPROVED** [review]
